tests/e2e_http.py

```diff
@@ -25,7 +25,8 @@ def run(args):
         with infra.ccf.network(
             hosts, args.build_dir, args.debug_nodes, args.perf_nodes, pdb=args.pdb
         ) as network:
-            primary, _ = network.start_and_join(args)
+            network.start_and_join(args)
+            primary, _ = network.find_primary()
```

Comment:
Seems weird that find_primary would return more than one node?

Reply:
I think this should be `find_nodes()`, as in the other tests, rather than `find_primary()`?

Reply:
This actually returns the term in which the `primary` was found. I'll make it more explicit.

Reply:
Why does it do that? It shouldn't be called find_primary() then.